src/app/components/sidebar/sidebar_item.rs

```diff
@@ -0,0 +1,168 @@
+use gettextrs::gettext;
+use glib::Properties;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+use crate::app::models::PlaylistSummary;
+
+const LIBRARY: &str = "library";
+const SAVED_TRACKS: &str = "saved_tracks";
+const NOW_PLAYING: &str = "now_playing";
+const SAVED_PLAYLISTS: &str = "saved_playlists";
+const PLAYLIST: &str = "playlist";
+pub const SAVED_PLAYLISTS_SECTION: &str = "saved_playlists_section";
+pub const CREATE_PLAYLIST_ITEM: &str = "create_playlist";
+
+#[derive(Debug)]
+pub enum SidebarDestination {
+    Library,
+    SavedTracks,
+    NowPlaying,
+    SavedPlaylists,
+    Playlist(PlaylistSummary),
+}
+
+impl SidebarDestination {
+    pub fn id(&self) -> &'static str {
+        match self {
+            Self::Library => LIBRARY,
+            Self::SavedTracks => SAVED_TRACKS,
+            Self::NowPlaying => NOW_PLAYING,
+            Self::SavedPlaylists => SAVED_PLAYLISTS,
+            Self::Playlist(_) => PLAYLIST,
+        }
+    }
+
+    pub fn title(&self) -> String {
+        match self {
+            // translators: This is a sidebar entry to browse to saved albums.
+            Self::Library => gettext("Library"),
+            // translators: This is a sidebar entry to browse to saved tracks.
+            Self::SavedTracks => gettext("Saved tracks"),
+            // translators: This is a sidebar entry to browse to saved playlists.
+            Self::NowPlaying => gettext("Now playing"),
+            // translators: This is a sidebar entry that marks that the entries below are playlists.
+            Self::SavedPlaylists => gettext("Playlists"),
+            Self::Playlist(PlaylistSummary { title, .. }) => title.clone(),
+        }
+    }
+
+    pub fn icon(&self) -> &'static str {
+        match self {
+            Self::Library => "library-music-symbolic",
+            Self::SavedTracks => "starred-symbolic",
+            Self::NowPlaying => "music-queue-symbolic",
+            Self::SavedPlaylists => "view-app-grid-symbolic",
+            Self::Playlist(_) => "playlist2-symbolic",
+        }
+    }
+}
+
+impl SidebarItem {
+    pub fn for_destination(dest: SidebarDestination) -> Self {
```

Comment:
I would name this `from_destionation` to be more consistent with rust naming conventions